### PR TITLE
Delete unused sagas/reducers for network graph.

### PR DESCRIPTION
**Pull request by dvail**
## Description

This PR is the result of a discovery by @pedrottimark that clicking a node in the Network Graph causes two sets of network requests to be sent to both `/deployments/:id` and `/networkpolicies/:pid`. Each set of requests originates from the following cause:
~~[a `useEffect` in the Network <Page> component](https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/src/Containers/Network/Page.js#L91) (used to enable the namespace of a deployment when visiting from the Risk page)~~ It turns out this request was unrelated.
- A click on a Cytoscape node, that dispatches a FETCH_DEPLOYMENT.request action to Redux

In looking for a way to remove the duplicate requests I found that the above ends up storing data that is unused, and that much of the reducer/action/saga plumbing it flows through is deprecated dead code.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added

## Testing Performed

If there are additional areas that should be manually tested that I did not account for below, please give me a heads up and I will check them as well.

I followed the deprecated flow in the following order:
- [a node click event](https://github.com/stackrox/stackrox/pull/1271/files#diff-144cd6c670b6b13a7cc59d07defe2e153b976aacf39f65446eccf1f2c36842c7L112) dispatches the initial `fetchDeployment.request` action that was created [here](https://github.com/stackrox/stackrox/pull/1271/files#diff-c11ed6479fb56727b2f3f66d0898db4a8b12fe1b773c75765437dc79584cd241L27)
- This action is picked up by the [NetworkSaga](https://github.com/stackrox/stackrox/pull/1271/files#diff-165272c1231ca7d39e1f7ce98c4e0cdb9f85a52de5e755471a68186667376288L294) that in turn calls a `getDeployment` saga from [DeploymentSagas](https://github.com/stackrox/stackrox/pull/1271/files#diff-42ad6572e73ab66fa03c4bfcb7a975723671fdac63a7d77aa44a8f101600d792L22)
- A search for the `getDeployment` saga reveals that it is only imported and used in the NetworkSaga file and can be deleted.
- The deleted `getDeployment` saga called the final instance of [this service function](https://github.com/stackrox/stackrox/pull/1271/files#diff-c7429905b7316ce086314b9fd82cc0f07e6cf1777080d22ebceb0ba825b10bb6L141), which can also be deleted
- The above flow is only used in [this reducer](https://github.com/stackrox/stackrox/pull/1271/files#diff-67e56bd97af5d59480b6cd11a17b1654dd529d8d9e73399ed4b8cc5974579fd4L103) when it detects the `FETCH_DEPLOYMENT.SUCCESS` action type.
- The state object returned by the above reducer is available through the [`getNodeDeployment` selector,](https://github.com/stackrox/stackrox/pull/1271/files#diff-67e56bd97af5d59480b6cd11a17b1654dd529d8d9e73399ed4b8cc5974579fd4L139) which is not called anywhere in the code base.

Key imports and search strings that are not available anywhere else in the code:
- getDeployment (the one called from deploymentSagas.js at least)
- fetchDeployment.request, fetchDeployment.success, fetchDeployment.failure
- FETCH_DEPLOYMENT (the action type, separate from FETCH_DEPLOYMENT**S**)

Tested that visiting the Network Graph via a Risk Page deployment works as expected:
image

Tested that visiting the Network Graph directly and clicking a node in the visualization opens the sidebar as expected:
image

Tested that a single request is sent to the  deployments API when visiting the Network Graph with a deployment id in the URL: (Note that this removes _both_ of the duplicate `v1/deployments/:id` calls when clicking a node, but _neither_ of the `v1/networkpolicies/:pid` calls)
image

### CODE REVIEW

File: ui/apps/platform/cypress/helpers/networkGraph.js [hunk 1]
```diff
@@ -30,9 +30,7 @@ export function clickOnNodeByName(cytoscape, node) {
 }
 
 export function clickOnDeploymentNodeByName(cytoscape, name) {
-    cy.intercept('GET', api.network.deployment).as('getDeployment');
     clickOnNodeByName(cytoscape, { type: 'DEPLOYMENT', name });
-    cy.wait('@getDeployment');
 }
```
Comment (pedrottimark) [inline ui/apps/platform/cypress/helpers/networkGraph.js:34]: Aha, as a follow up for me after this is merged to undo part of what I did in #1239
* replace `clickOnDeploymentNodeByName` calls
* with `clickOnNameByName`

**Review by pedrottimark — APPROVED** [review]
Super! Thank you very much.

File: ui/apps/platform/cypress/helpers/networkGraph.js (hunk 1 above)
Comment (dvail) [reply ui/apps/platform/cypress/helpers/networkGraph.js:34]: Ah wow I didn't realize this code was something you just added.

I am hopeful that the code removal here doesn't end up causing e2e test failures. Even though the network requests were unneeded, they did impact timing in Cypress tests. Everything was passing for me locally, so hopefully that continues to be the case in CI.

Comment (pedrottimark) [reply ui/apps/platform/cypress/helpers/networkGraph.js:34]: Running this branch, there is a /v1/deployments/:id request for the **first** click on a deployment node.

So the following intercept, click, wait sandwich works, although I will verify as follow up whether it is needed:
https://github.com/stackrox/stackrox/blob/master/ui/apps/platform/cypress/integration/networkGraph/networkGraph.test.js#L20-L27

Comment (pedrottimark) [reply ui/apps/platform/cypress/helpers/networkGraph.js:34]: Not unique to this pull request, I see some errors when using browser history to go back to main dashboard.
back

**Review by sachaudh — APPROVED** [review]

**Pull request merged**